README.md

```diff
@@ -0,0 +1,116 @@
+# Microsoft Authentication Library (MSAL) for Python
+
+The MSAL library for Python gives your app the ability to begin using the [Microsoft Cloud](https://cloud.microsoft.com)
+by supporting [Microsoft Azure Active Directory](https://azure.microsoft.com/en-us/services/active-directory/)
+and [Microsoft Accounts](https://account.microsoft.com) in a converged experience using industry standard OAuth2 and OpenID Connect.
+Soon MSAL Python will also support [Azure AD B2C](https://azure.microsoft.com/services/active-directory-b2c/).
+
+More and more detail about MSAL Python functionality and usage will be documented in the
+[Wiki](https://github.com/AzureAD/microsoft-authentication-library-for-python/wiki).
+
+## Installation
+
+1. If you haven't already, [install and/or upgrade the pip](https://pip.pypa.io/en/stable/installing/)
+   of your Python environment to a recent version. We tested with pip 18.1.
+2. For now, you can install from our latest dev branch, by `pip install https://github.com/AzureAD/microsoft-authentication-library-for-python.git@dev`
+
+## Usage
+
+Before using MSAL Python (or any MSAL SDKs, for that matter), you will have to
+[register your application with the AAD 2.0 endpoint](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-v2-register-an-app).
+
+Acquiring tokens with MSAL Python is somewhat different than ADAL Python. You will need to follow this 3-step pattern.
+
+1. Contrary to ADAL (which proposes the notion of `AuthenticationContext`, which is a connection to Azure AD),
+   MSAL proposes a clean separation between
+   [public client applications, and confidential client applications](https://tools.ietf.org/html/rfc6749#section-2.1).
+   So you will first create either a `PublicClientApplication` or a `ConfidentialClientApplication` instance,
+   and ideally reuse it during the lifecycle of your app. For example:
+
+   ```python
+   from msal import PublicClientApplication
+   app = PublicClientApplication("your_client_id", authority="...")
+   ```
+   
+   Later, each time you would want an access token, you start by:
+   ```python
+   result = None
+   ```
+
+2. The API model in MSAL provides you explicit control on how to utilize token cache.
```

Comment:
At the end of this step as an optional, can we show or explain how the developer can explicitly control the token cache instead of using MSAL cache if they choose to?

Reply:
I think the content in readme should be a minimalist on how to obtain a token. Anything beyond that, such as "How to customize cache", is definitely something needed to be documented - in its own wiki page. :-)